src/main/java/org/mitre/dsmiley/httpproxy/ProxyServlet.java

```diff
@@ -650,6 +651,15 @@ protected void copyResponseEntity(HttpResponse proxyResponse, HttpServletRespons
               os.flush();
             }
           }
+        } catch (IOException | RuntimeException ex) {
+          // A chunked response is read to completion when #close is called.
+          // If the sending site does not timeout or keeps sending, the
+          // connection will be kept open indefinitely. Closing the respone
+          // object terminates the stream and allows cleanly closing the
+          // inputstream
+          if(proxyResponse instanceof Closeable) {
```

Comment:
I'm glad this now only happens when an exception is thrown, but I think it could be improved a little by moving the logic into handleRequestException() or perhaps immediately prior to where that's called.  It will re-throw the exception as well, and I see here you forgot that.  It'll also end up closing this even without chunked responses -- which makes sense to me as I don't see why we need to make an exception.

Reply:
I had to reread `#service` and have to agree. That also allows to simplify the `copyResponseEntity` method, as `service` takes care of consuming the entity. The finally block in `copyResponseEntity` can be thus removed and the closing of the http response can indeed be moved to the `handleRequestException` method.

I updated the PR accordingly. The unittest still come back cleanly.